File: auth-service/src/main/java/io/simakkoi9/authservice/model/Gender.java
package io.simakkoi9.authservice.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum Gender {
    MALE(1),
    FEMALE(2);

    private final int code;

    Gender(int code) {
        this.code = code;
    }

    public int getCode() {
        return code;
    }

    @JsonCreator
    public static Gender fromValue(String value) {
        for (Gender gender : values()) {
            if (gender.name().equalsIgnoreCase(value)) {
                return gender;
            }
        }
        throw new IllegalArgumentException("Unknown gender value: " + value);
    }

    public static Gender fromCode(int code) {
        for (Gender gender : values()) {
            if (gender.code == code) {
                return gender;
            }
        }
        throw new IllegalArgumentException("Unknown gender code: " + code);
    }

    @JsonValue
    public String toValue() {
        return this.name();
    }
} 

File: driver-service/src/main/kotlin/io/simakkoi9/driverservice/model/entity/Gender.kt
package io.simakkoi9.driverservice.model.entity

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import io.simakkoi9.driverservice.util.ErrorMessages.UNKNOWN_GENDER_CODE
import io.simakkoi9.driverservice.util.ErrorMessages.UNKNOWN_GENDER_VALUE

enum class Gender(val code: Int) {
    MALE(1),
    FEMALE(2);

    companion object {
        @JvmStatic
        @JsonCreator
        fun fromValue(value: String): Gender {
            return entries.find {
                it.name.equals(value, ignoreCase = true)
            } ?: throw IllegalArgumentException(UNKNOWN_GENDER_VALUE.format(value))
        }

        @JvmStatic
        fun fromCode(code: Int): Gender {
            return entries.find {
                it.code == code
            } ?: throw IllegalArgumentException(UNKNOWN_GENDER_CODE.format(code))
        }
    }

    @JsonValue
    fun toValue(): String {
        return this.name
    }
}